### PR TITLE
Added savepdf Command.

### LINKED ISSUE #123: Save as pdf command to display tasks of each driver for that day.
Command:
format: savepdf

save a pdf that includes the tasks for each driver for the day.

### PR DESCRIPTION
**Pull request by junnbang**
Closes #123

Format:
-savepdf
-savepdf [DATE]

Saves drivers' tasks into pdf format.

image

### CODE REVIEW

File: src/main/java/seedu/address/logic/parser/ParserUtil.java [hunk 1]
```diff
@@ -166,6 +167,20 @@ public static LocalDate parseDate(String date) throws ParseException {
         return getDate(trimmedDate);
     }
 
+    /**
+     * Parses a {@code String date} into an {@code LocalDate}.
+     *
+     * @throws ParseException if date format is invalid.
+     */
+    public static LocalDate parseAnyDate(String date) throws ParseException {
```
Comment (tysng) [inline src/main/java/seedu/address/logic/parser/ParserUtil.java]: This should replace `parseDate`. Parser shouldn't be checking if the day is before today actually, it's a parser so any date that is valid should pass.

File: src/main/java/seedu/address/logic/parser/SavePdfCommandParser.java [hunk 2]
```diff
@@ -0,0 +1,35 @@
...
+     * @throws ParseException if the user input does not conform the expected format
+     */
+    public SavePdfCommand parse(String args) throws ParseException {
+        String trimmedArgs = args.trim();
+        if (trimmedArgs.isEmpty()) {
+            return new SavePdfCommand(Optional.empty());
+        }
+
+        try {
+            Optional<LocalDate> date = Optional.of(ParserUtil.parseAnyDate(args));
```
Comment (tysng) [inline src/main/java/seedu/address/logic/parser/SavePdfCommandParser.java]: Any reason with using Optional? Date should always be non null.

File: src/main/java/seedu/address/model/pdfmanager/PdfWrapperLayout.java [hunk 3]
```diff
@@ -0,0 +1,157 @@
...
+
+    /**
+     * Filter task list to get tasks on a specific date. Only account for ONGOING and COMPLETE tasks.
+     * INCOMPLETE tasks are not needed as they are not assigned to any drivers.
+     *
+     * @param tasks task list.
+     * @param date  date of delivery.
+     * @return filtered task list that contains only tasks on a specific date and are ONGOING or COMPLETE.
+     */
+    private List<Task> filterTasksBasedOnDate(List<Task> tasks, LocalDate date) {
```
Comment (tysng) [inline src/main/java/seedu/address/model/pdfmanager/PdfWrapperLayout.java]: This code seems very familiar. If it's anywhere else, consider reusing that code.

**Review by tysg — COMMENTED** [review]
Seems like a pretty cool feature. Just some adjustments

File: src/main/java/seedu/address/logic/commands/SavePdfCommand.java [hunk 4]
```diff
@@ -0,0 +1,65 @@
...
+            + "Saves all drivers' tasks into PDF for the specific date. "
+            + "Date field is optional. "
+            + "if a date is not specified, then today's date will be chosen. \n"
+            + "Parameters: "
+            + "[DATE] \n"
+            + "Example: " + COMMAND_WORD + " 20/10/2019";
+
+    private static final String FILE_PATH_FOR_PDF = "./data/" + FILE_NAME + " %1$s.pdf";
+
+    private final Clock clock;
```
[the remark this answers is not in the document]
Comment (junnbang) [reply src/main/java/seedu/address/logic/commands/SavePdfCommand.java]: solved

File: src/main/java/seedu/address/logic/parser/ParserUtil.java (hunk 1 above)
Comment (junnbang) [reply src/main/java/seedu/address/logic/parser/ParserUtil.java]: okay, check today on wards at command side

File: src/main/java/seedu/address/logic/parser/SavePdfCommandParser.java (hunk 2 above)
Comment (junnbang) [reply src/main/java/seedu/address/logic/parser/SavePdfCommandParser.java]: savepdf has optional date input, so if there's no date input, then ill throw in a optional.empty() as parameter of the constructor

File: src/main/java/seedu/address/model/pdfmanager/PdfWrapperLayout.java (hunk 3 above)
Comment (junnbang) [reply src/main/java/seedu/address/model/pdfmanager/PdfWrapperLayout.java]: i try and find

**Review by tysg — APPROVED** [review]

**Pull request merged**